### ISSUE TITLE
Minimum JRE version is 1.8 instead of 1.6

### ISSUE THREAD

**Issue by heidivanparys** [ShapeChange/ShapeChange#43]
In commit ef975d78bc8ea0727006f3440fa502e2a6950f56 , the Jena version was changed from 2.11.1 to 3.1.0 . Jena requires Java 8, see https://jena.apache.org/download/index.cgi .

This means that the minimum JRE version for ShapeChange should be changed from 1.6 to 1.8. This would include:

- updating ShapeChange/pom.xml 
- updating  ShapeChange/src/test/java/de/interactive_instruments/ShapeChange/TestInstance.java 
- updating the documentation on http://shapechange.net/get-started/

I noticed this issue when running the tests with a 1.6 JRE, I got the following error:
```java
java.lang.UnsupportedClassVersionError: org/apache/jena/rdf/model/Model : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:191)
	at de.interactive_instruments.ShapeChange.Converter.resetSingleTargets(Converter.java:773)
	at de.interactive_instruments.ShapeChange.Converter.executeTargets(Converter.java:537)
	at de.interactive_instruments.ShapeChange.Converter.convert(Converter.java:240)
	at de.interactive_instruments.ShapeChange.Converter.convert(Converter.java:132)
	at de.interactive_instruments.ShapeChange.TestInstance.init(TestInstance.java:87)
	at de.interactive_instruments.ShapeChange.TestInstance.<init>(TestInstance.java:59)
	at de.interactive_instruments.ShapeChange.BasicTest.multiTest(BasicTest.java:575)
	at de.interactive_instruments.ShapeChange.BasicTest.test(BasicTest.java:406)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Comment by jechterhoff** [comment]
Thank you for bringing this up.
It's true that the new ontology target requires Java 8 to execute (this is [documented online](http://shapechange.net/targets/ontology/uml-rdfowl-based-isois-19150-2/)) - and thus also the execution of our Unit tests.
However, if that target is not executed, then ShapeChange can be run with a lower version of Java.
We tried to be backwards-compatible with regards to Java, to support those users that cannot easily switch to the most recent version of Java. On the other hand, security updates etc. are now only publicly available for the most recent Java version.
I'm open for suggestions and a discussion.

**Comment by heidivanparys** [comment]
I understand, it's of course a valid requirement that Java 6 should be supported as long as possible.

I would suggest to add a note to the Developer Documentation. It currently describes that Java 6 is required, whereas the runtime JRE for the unit tests should be Java 8.

**Comment by jechterhoff** [comment]
Good point. I've updated the developer documentation.

**Comment by jechterhoff** [comment]
Update: We've decided to move towards Java 8, for the development of the next release.
Reasons:
* New Java language features and APIs help the development process.
* Users that still use Java installations for which the "End of Public Updates" has passed (like for Java 6 and 7) should really move to a Java version for which public updates are still available/created (like Java 8).

**Comment by cportele** [comment]
Closed since release 2.4.0 requires Java 1.8.
